syntax = "proto3";

package flowpipe.stages.sqs.source.v1;

// Configuration for SQSSource stage
message SQSSourceConfig {
  // SQS queue URL
  string queue_url = 1;

  // AWS region (e.g., us-east-1)
  string region = 2;

  // Optional endpoint override
  string endpoint = 3;

  // Optional AWS access key ID
  string access_key_id = 4;

  // Optional AWS secret access key
  string secret_access_key = 5;

  // Optional AWS session token
  string session_token = 6;

  // Maximum number of messages to receive per request (1-10)
  int32 max_number_of_messages = 7;

  // Long poll wait time in seconds (0-20)
  int32 wait_time_seconds = 8;

  // Visibility timeout in seconds
  int32 visibility_timeout_seconds = 9;

  // Delete message after it is received
  bool delete_after_read = 10;
}
